api-consumption/src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -81,16 +82,24 @@ public void streamEvents(final AtomicBoolean connectionReady, final Runnable che
             long lastKpiEventSent = System.currentTimeMillis();
             long bytesInMemory = 0;
 
-            while (connectionReady.get() &&
-                    !blacklistService.isConsumptionBlocked(config.getEtName(), config.getConsumingClient()
-                            .getClientId())) {
+            while (connectionReady.get()) {
+
+                if (eventStreamChecks.isConsumptionBlocked(
+                        Collections.singleton(config.getEtName()),
+                        config.getConsumingClient().getClientId())) {
+                    break;
+                }
 
                 checkAuthorization.run();
 
                 if (consumedEvents.isEmpty()) {
-                    // TODO: There are a lot of optimizations here, one can significantly improve code by processing
-                    // all events at the same time, instead of processing one by one.
-                    consumedEvents.addAll(eventConsumer.readEvents());
+                    final List<ConsumedEvent> eventsFromKafka = eventConsumer.readEvents();
+                    for (final ConsumedEvent evt: eventsFromKafka) {
+                        if (eventStreamChecks.isConsumptionBlocked(evt)) {
```

Comment:
I expected to see this code in common place like NakadiKafkaConsumer, where the filtering could be done in one place instead of putting it in 2 places (lola, hila).

Reply:
Thanks for pointing out! Will work on it.

Reply:
After spending some time refactoring to this approach, it seems that it is not working, and is completely wrong. The reason is that we may be using several data providers (not only kafka), they all should be working, and we can not use it. 
Event thought we have only KAFKA inplace, organization of code enforces permissions check on a different level, as for example filtering on NakadiKafkaConsumer level will lead to timelines being broken for some cases.
Therefore I propose not to do this move.

Reply:
@adyach